src/Provider/ConstantProviderFactory.php

```diff
@@ -8,48 +8,51 @@
  * For the full copyright and license information, please view the file license.txt that was distributed with this source code.
  */
 
-declare(strict_types = 1);
+declare(strict_types=1);
 
 namespace Kdyby\DateTimeProvider\Provider;
 
 use DateTime;
 use DateTimeImmutable;
 use DateTimeZone;
+use Kdyby\DateTimeProvider\NotImplementedException;
+use Kdyby\StrictObjects\Scream;
+use function date_default_timezone_get;
+use function get_class;
+use function gettype;
+use function is_numeric;
+use function is_object;
+use function is_string;
+use function sprintf;
 
 /**
  * Helper factory to create ConstantProvider from arbitrary input.
  */
 class ConstantProviderFactory
 {
-
-	use \Kdyby\StrictObjects\Scream;
-
-	/**
-	 * @param string|int|\DateTimeImmutable|\DateTime $dateTime
-	 */
-	public function create($dateTime): ConstantProvider
-	{
-		if ($dateTime instanceof DateTimeImmutable) {
-			return new ConstantProvider($dateTime);
-
-		} elseif ($dateTime instanceof DateTime) {
-			return new ConstantProvider(DateTimeImmutable::createFromMutable($dateTime));
-
-		} elseif (is_numeric($dateTime)) {
-			return new ConstantProvider((new DateTimeImmutable(sprintf('@%.6f', $dateTime)))->setTimezone(new DateTimeZone(date_default_timezone_get())));
-
-		} elseif (is_string($dateTime)) {
-			throw new \Kdyby\DateTimeProvider\NotImplementedException(sprintf(
-				'Cannot process datetime in given format "%s"',
-				$dateTime
-			));
-
-		} else {
-			throw new \Kdyby\DateTimeProvider\NotImplementedException(sprintf(
-				'Cannot process datetime from given value %s',
-				is_object($dateTime) ? get_class($dateTime) : gettype($dateTime)
-			));
-		}
-	}
-
+    use Scream;
+
+    /**
+     * @param string|int|\DateTimeImmutable|\DateTime $dateTime
+     */
+    public function create($dateTime) : ConstantProvider
+    {
+        if ($dateTime instanceof DateTimeImmutable) {
+            return new ConstantProvider($dateTime);
+        } elseif ($dateTime instanceof DateTime) {
+            return new ConstantProvider(DateTimeImmutable::createFromMutable($dateTime));
+        } elseif (is_numeric($dateTime)) {
+            return new ConstantProvider((new DateTimeImmutable(sprintf('@%.6f', $dateTime)))->setTimezone(new DateTimeZone(date_default_timezone_get())));
+        } elseif (is_string($dateTime)) {
+            throw new NotImplementedException(sprintf(
+                'Cannot process datetime in given format "%s"',
+                $dateTime
+            ));
+        } else {
```

Comment:
This is actually a useless else. I'm a bit surprised that Doctrine CS doesn't have a rule for that.

Reply:
Actually whole if/elseif/else chain is useless here, there is EarlyExit sniff, but doesn't catch this, maybe @kukulich will know why exactly?

Reply:
That's because the sniff ignores `if-else` with `elseif`.

This should really be covered but it's not easy... I will try to support this (probably without the fixer).

Reply:
@kukulich Understood, no problem, imho better not catching these than false positives. 👍

Reply:
(May probably be better suited for PHPStan?)

Reply:
The check is relatively simple but it's difficult to report all five errors at once (and even more difficult to fix all of them at once).